controllers/machinehealthcheck_controller_test.go

```diff
@@ -0,0 +1,428 @@
+			}
+			Expect(k8sClient.Create(ctx, mhcToCreate)).To(Succeed())
+
+			Eventually(func() []metav1.OwnerReference {
+				mhc := &clusterv1.MachineHealthCheck{}
+				err := k8sClient.Get(ctx, client.ObjectKey{Namespace: mhcToCreate.GetNamespace(), Name: mhcToCreate.GetName()}, mhc)
+				if err != nil {
+					return []metav1.OwnerReference{}
+				}
+				return mhc.GetOwnerReferences()
+			}, timeout).Should(ConsistOf(ortc.expected()))
+		},
+		Entry("when no existing owner references exist", ownerReferenceTestCase{
+			original: []metav1.OwnerReference{},
+			expected: func() []metav1.OwnerReference {
+				return []metav1.OwnerReference{ownerReferenceForCluster(ctx, testCluster)}
+			},
+		}),
+		Entry("when modifying existing owner references", ownerReferenceTestCase{
+			original: []metav1.OwnerReference{{Kind: "Foo", APIVersion: "foo.bar.baz/v1", Name: "Bar", UID: "12345"}},
+			expected: func() []metav1.OwnerReference {
+				return []metav1.OwnerReference{{Kind: "Foo", APIVersion: "foo.bar.baz/v1", Name: "Bar", UID: "12345"}, ownerReferenceForCluster(ctx, testCluster)}
+			},
+		}),
+	)
+
+})
+
+func cleanupTestMachineHealthChecks(ctx context.Context, c client.Client) error {
+	mhcList := &clusterv1.MachineHealthCheckList{}
+	err := c.List(ctx, mhcList)
+	if err != nil {
+		return err
+	}
+	for _, mhc := range mhcList.Items {
+		m := mhc
+		err = c.Delete(ctx, &m)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func ownerReferenceForCluster(ctx context.Context, c *clusterv1.Cluster) metav1.OwnerReference {
+	// Fetch the cluster to populate the UID
+	cc := &clusterv1.Cluster{}
+	Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: c.GetNamespace(), Name: c.GetName()}, cc)).To(Succeed())
+
+	return metav1.OwnerReference{
+		APIVersion: clusterv1.GroupVersion.String(),
+		Kind:       "Cluster",
+		Name:       cc.Name,
+		UID:        cc.UID,
+	}
+}
+
+func TestClusterToMachineHealthCheck(t *testing.T) {
+	// This test sets up a proper test env to allow testing of the cache index
+	// that is used as part of the clusterToMachineHealthCheck map function
+
+	// BEGIN: Set up test environment
+	g := NewWithT(t)
+
+	testEnv = &envtest.Environment{
+		CRDs: []*apiextensionsv1beta1.CustomResourceDefinition{
+			external.TestGenericBootstrapCRD,
+			external.TestGenericBootstrapTemplateCRD,
+			external.TestGenericInfrastructureCRD,
+			external.TestGenericInfrastructureTemplateCRD,
+		},
+		CRDDirectoryPaths: []string{filepath.Join("..", "config", "crd", "bases")},
+	}
+
+	var err error
+	cfg, err := testEnv.Start()
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cfg).ToNot(BeNil())
+	defer func() {
+		g.Expect(testEnv.Stop()).To(Succeed())
+	}()
+
+	g.Expect(clusterv1.AddToScheme(scheme.Scheme)).To(Succeed())
+
+	mgr, err = manager.New(cfg, manager.Options{
+		Scheme:             scheme.Scheme,
+		MetricsBindAddress: "0",
+		NewCache: func(config *rest.Config, opts cache.Options) (cache.Cache, error) {
+			syncPeriod := 1 * time.Second
```

Comment:
What's the motivation for having a sync period?

Reply:
Honestly, I copied the manager set up over from [suite_test](https://github.com/kubernetes-sigs/cluster-api/blob/1680c04e2b4cfee433aab19f3dd89bc4b1102f36/controllers/suite_test.go#L97-L105) assuming there was some good reason for it 😅 I don't think it will make a difference to these tests and will get rid of it